### PR TITLE
Remove get  for getters

### PR DESCRIPTION
**Pull request by fengalin**
This is an attempt at fixing https://github.com/gtk-rs/gir/issues/963 & https://github.com/gtk-rs/gir/issues/962.

The changes are fairly minimal because most of the "hard work" is done in [`fix-getters-rules`](https://github.com/fengalin/fix-getters/tree/main/rules). I decided to implement those rules in a separate crate because they are not `gir` specific:

* They can be used for non-`gir` based crates.
* They are required for manual code update.

However, I took good care at keeping this dependency minimal. The package used in `gir` only pulls in `once_cell` which is already pulled in by `gir`. I also introduced a feature to remove a module which makes sense in this package, but which is of no use for `gir`.

Please take a look at the `README`s in [`fix-getters`](https://github.com/fengalin/fix-getters). They explain the assumptions and decisions I had to make to achieve this. Of course, these assumption are all subject to debate.

Changes to `gtk-rs` can be found in [this PR](https://github.com/gtk-rs/gtk-rs/pull/211).

Please see the request in [this PR](https://github.com/gtk-rs/gtk-rs/pull/211).

Fixes https://github.com/gtk-rs/gir/issues/963

### CODE REVIEW

**Comment by fengalin** [comment]
Of course, CI failure is expected.

**Comment by sdroege** [comment]
@GuillaumeGomez What do you think? :)

File: README.md
Comment (GuillaumeGomez) [inline README.md:480]: That's exactly what I'm afraid of in the end... Two questions about this:

1. How much name conflicts would be there?
2. Couldn't we automatically detect those conflicts?

Comment (fengalin) [reply README.md:480]: > How much name conflicts would be there?

See https://github.com/gtk-rs/gtk-rs/pull/211/commits/4fb1e7b4b61f3ada35cee9988a58b9a3b48639aa & https://github.com/gtk-rs/gtk-rs/pull/211/commits/021ee0930b9cd7dd4448fa3647a095c0334e70e2. Please take a look at https://github.com/gtk-rs/gtk-rs/pull/211.

> Couldn't we automatically detect those conflicts?

That's already been done for `get_property_...`, others need some kind of human cognition IMHO.

**Comment by fengalin** [comment]
This is ready. See https://github.com/gtk-rs/gtk-rs/pull/211 for an application on 2 hours ago's `gtk-rs` master.

**Important: when this is merge, other projects will not be able to regen before the `gtk-rs` PR is merged too, otherwise it will break everything up.**

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
So just to sum things up for others: it could create conflicts between a trait method and a type method (even if unlikely). But that's pretty much it. For such cases, a nice thing could be to add a lint in clippy but that is not a blocker.

Waiting for @slomo or @bilelmoussaoui approval and then let's merge!

File: Cargo.toml
Comment (sdroege) [inline Cargo.toml:16]: This should probably get a better name so it can go on crates.io without looking strange :) Any suggestions?

Comment (sdroege) [reply Cargo.toml:16]: Not a blocker for merging this now though, just before the release.

File: src/analysis/functions.rs
Comment (sdroege) [inline src/analysis/functions.rs:566]: `new_for` would be another pattern btw

File: Cargo.toml
Comment (fengalin) [reply Cargo.toml:16]: The actual crate name is `fix-getters-rules`, is that better?

**Review by sdroege — COMMENTED** [review]
Apart from these comments here and the ones on the gtk-rs PR, this looks all good to me. Good work! :)

File: Cargo.toml
Comment (sdroege) [reply Cargo.toml:16]: Yeah

**Comment by fengalin** [comment]
This is ready for merge now I think. See gtk-rs/gtk-rs#211 for an application on lastest gtk-rs's master.

**Important: when this is merged, please wait until I update gtk-rs/gtk-rs#211 with the merged Gir before merging anything else that could perform a regen on `gtk-rs/gtk-rs`.**

**Comment by fengalin** [comment]
@sdroege @bilelmoussaoui this PR needs your approvals. Then we can merge!

**Review by bilelmoussaoui — APPROVED** [review]

**Pull request merged**
